### PR TITLE
Week 9 / V1.2 Documentation updates

### LINKED ISSUE #46: Week 9: Update UG - Features: Sort Contacts

### LINKED ISSUE #47: Week 9: Update UG - Command Summary: Sort

### LINKED ISSUE #48: Week 9: Update UG - Features: Viewing help - Update Screenshot

### LINKED ISSUE #49: Week 9: Update UG - Features: Editing the data file
Replace mentions of AddressBook with SoConnect

### LINKED ISSUE #50: Week 9: Update UG - Features: Saving the data
Replace mentions of AddressBook with SoConnect

### LINKED ISSUE #51: Week 9: Update UG - Quick start - Update Screenshot

### LINKED ISSUE #69: Week 9: Update DG - Implementation: Sorting
Add Implementation Description
Check User Story to see if updates are needed
Check Use Case to see if updates are needed

### PR DESCRIPTION
**Pull request by Eugene-Ong-W-X**
1. Sort Feature updated in UG (Feature section & Command Summary)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+Step 1. The user executes `sort t/!friend n/` command to perform a multi-level sort. `SortCommandParser` calls `ArgumentTokenizer#tokenizeToList()` to separate the parameters of `t/!friend` and `n/`.
+
+Step 2. The `sort` command sorts the currently displayed list by name first, calling `Model#sortByName(Boolean isReverse)` where `isReverse = false`.
+
+Step 3. The `sort` command sorts the currently displayed list by the `friend` tag next, calling `Model#sortByTag(Tag tag, Boolean isReverse)` where `isReverse = true`.
+
+The following sequence diagram shows how the sort operation works:
+
+(insert sequence diagram here)
```
Comment (florentianayuwono) [inline docs/DeveloperGuide.md:184]: For now this dummy placeholder is okay?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +154,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+</div>
+
+Step 4. The user is shown the sorted list. The sorted list contains the same contacts as the previous displayed list. It has two sections, the first section contains contacts without the `friend` tag and the second section contains contacts with the `friend` tag. Each section is sorted by name in alphabetical order.
+
+The following activity diagram summarizes what happens when a user executes a sort command:
+
+(insert activity diagram here)
+
+#### Design considerations:
```
Comment (florentianayuwono) [inline docs/DeveloperGuide.md]: Maybe can be `Design consideration`, without `s:`. My concern is it doesn't look quite good in the Table of Contents.

image

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -154,6 +154,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+* **Alternative 2:** Single complex sorting method that combines all the individual sorting for each parameter.
+    * Pros: Save time as only 1 sorting operation is carried out.
+    * Cons: Harder to modify when more parameters are added. Can result in more bugs due to complexity.
+
+_{more aspects and alternatives to be added}_
+
 ### \[Proposed\] Undo/redo feature
 
 #### Proposed Implementation
```
Comment (florentianayuwono) [inline docs/DeveloperGuide.md]: `Proposed implementation`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -427,12 +480,12 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
       Use case resumes at step 3.
 
-**Use case: sort by name**
+**Use case: Sort by name**
 
 **MSS**
 
-1.  User request sort by name
-2.  SoConnect sorts the list by name.
+1.  User enters command to sort the contact list by name
```
Comment (florentianayuwono) [inline docs/DeveloperGuide.md]: Add `.`

File: docs/UserGuide.md [hunk 5]
```diff
@@ -97,20 +97,21 @@ Format: `list`
...
 
 Default sorting orders:
 * Alphabetical order for *names*, *emails*, *addresses*.
 * Increasing order for *phone numbers*.
-* Sorts contacts that have a specified *tag* before those without the *tag*.
+* Contacts that have a specified *tag* appear before those without the *tag*.
 
 Format: `sort [n/] [p/] [e/] [a/] [t/TAG]…​`
-* Add `!` in front of an attribute to sort in reverse order.
+* To sort in reverse order, insert a `!` like so `[n/!] [p/!] [e/!] [a/!] [t/!TAG]`.
```
Comment (florentianayuwono) [inline docs/UserGuide.md]: The `like so` a bit awkward.

**Review by florentianayuwono — COMMENTED** [review]
Great work, some suggestions attached. I case there is not enough time to make changes can just merge first then edit later.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Eugene-Ong-W-X) [reply docs/DeveloperGuide.md:184]: I put a placeholder here to remind myself to insert a sequence diagram there in v1.3, hahaha. I can remove it, if you think it's out of place?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Eugene-Ong-W-X) [reply docs/DeveloperGuide.md]: Sure!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (Eugene-Ong-W-X) [reply docs/DeveloperGuide.md]: Ok

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Eugene-Ong-W-X) [reply docs/DeveloperGuide.md]: Alright, I'll make the changes here

File: docs/UserGuide.md (hunk 5 above)
Comment (Eugene-Ong-W-X) [reply docs/UserGuide.md]: Hmm, I'll see if I can word it better

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (florentianayuwono) [reply docs/DeveloperGuide.md:184]: Nope I think it's good!

File: docs/UserGuide.md [hunk 6]
```diff
@@ -97,20 +97,21 @@ Format: `list`
...
 Default sorting orders:
 * Alphabetical order for *names*, *emails*, *addresses*.
 * Increasing order for *phone numbers*.
-* Sorts contacts that have a specified *tag* before those without the *tag*.
+* Contacts that have a specified *tag* appear before those without the *tag*.
 
 Format: `sort [n/] [p/] [e/] [a/] [t/TAG]…​`
-* Add `!` in front of an attribute to sort in reverse order.
+* To sort in reverse order, use these modified parameters: `[n/!] [p/!] [e/!] [a/!] [t/!TAG]`.
+* To sort with multiple parameters, arrange the parameters in order of decreasing priority.
```
Comment (florentianayuwono) [inline docs/UserGuide.md:109]: Nice!

**Review by florentianayuwono — APPROVED** [review]
LGTM

**Pull request merged**